shared/react-native/ios/KeybaseShare/FilesViewController.m

```diff
@@ -0,0 +1,222 @@
+    NSString* jsonFiles = KeybaseExtensionListPath(self.path, &error); // returns the path list in JSON format
+    if (jsonFiles == nil) {
+      dispatch_async(dispatch_get_main_queue(), ^{
+        NSLog(@"failed to get files: %@", error);
+        [av stopAnimating];
+        [av removeFromSuperview];
+      });
+      // just show blank in this case
+      return;
+    }
+    [self parseFiles:jsonFiles];
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [av stopAnimating];
+      [self.tableView reloadData];
+      [av removeFromSuperview];
+    });
+  });
+}
+
+NSInteger sortEntries(NSDictionary* one, NSDictionary* two, void* context) {
+  int t1 = [[one objectForKey:@"direntType"] intValue];
+  int t2 = [[two objectForKey:@"direntType"] intValue];
+  if (t1 == 1 && t2 != 1) {
+    // `one` is a folder and `two` isn't.
+    return NSOrderedAscending;
+  } else if (t2 == 1 && t1 != 1) {
+    // `two` is a folder and `one` isn't.
+    return NSOrderedDescending;
+  } else {
+    // Both are a folder or neither is.
+    return [one[@"name"] compare:two[@"name"]];
+  }
+}
+
+bool filterWritableEntries(NSDictionary* entry) {
+  return [entry objectForKey:@"writable"];
+}
+
+- (void)parseFiles:(NSString*)jsonFiles {
+  NSError* error = nil;
+  NSData* data = [jsonFiles dataUsingEncoding:NSUTF8StringEncoding];
+  NSArray* items = [NSJSONSerialization JSONObjectWithData:data options: NSJSONReadingMutableContainers error: &error];
+  NSArray* sortedAndFilteredItems;
+  if (!items) {
+    if (error) {
+      NSLog(@"parseFiles: error parsing JSON: %@", error);
+    }
+    // At least show an empty folder.
+    sortedAndFilteredItems = [[NSArray alloc] init];
+  } else {
+    // Sort items: directories first, then alphabetically.
+    sortedAndFilteredItems = [items sortedArrayUsingFunction:sortEntries context:NULL];
+    
+    // For paths deeper than 2, filter out folders that aren't writable.
+    unsigned long pathLength = [self pathLength];
+    if (pathLength > 2) {
+      sortedAndFilteredItems = [sortedAndFilteredItems filteredArrayUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(id entry, NSDictionary* bindings) {
+        return [entry objectForKey:@"writable"];
+      }]];
+    }
+  }
+  if ([self.path isEqualToString:@"/"]) {
+    // If we're at the root, don't show a back navigation item.
+    [self setDirectoryEntries:sortedAndFilteredItems];
+  } else {
+    // Otherwise, prepend a back navigation item.
+    NSMutableArray* itemsWithBack = [[NSMutableArray alloc] init];
+    [itemsWithBack addObject:[[NSDictionary alloc] initWithObjectsAndKeys:
+                              UpOneLevel, @"name",
+                              [NSNumber numberWithInt:1], @"direntType",
+                              nil]];
+    [itemsWithBack addObjectsFromArray:sortedAndFilteredItems];
+    [self setDirectoryEntries:itemsWithBack];
+  }
+}
+
+- (void)didReceiveMemoryWarning {
+  KeybaseExtensionForceGC();
```

Comment:
I wonder if it'd make sense to show a popup explaining that we've been warned about memory consumption and are trying to deal with it. This way if we crash eventually, user knows it's because of limited RAM rather than some random bug.

Reply:
i don't think we should do this. this can be caused by anything (other apps etc)

Reply:
We do this on the chat side of the extension to try to minimize the probability that we'll be OOM killed. I was just following that pattern. @mmaxim WDYT?

Reply:
Running this is fine here.